src/main/java/org/testng/internal/Parameters.java

```diff
@@ -52,11 +52,39 @@
       AfterMethod.class
   };
 
-  private static Map<String, List<Class>> mapping = Maps.newHashMap();
+  private static Map<String, List<Class<?>>> mapping = Maps.newHashMap();
+/*
+          +--------------+--------------+---------+--------+----------+-------------+
```

Comment:
@cbeust - Yes. This seems to be current behavior. Just to ensure that this was the case, I backtracked to using TestNG 6.9.9 and found that this was the case.

Here's a sample
```java
import org.testng.TestNG;
import org.testng.annotations.BeforeSuite;
import org.testng.annotations.Test;
import org.testng.xml.XmlTest;

import java.io.File;

public class SuiteLevelInjection {
    @BeforeSuite
    public void beforeSuite(XmlTest xmlTest) {
        String file = TestNG.class.getProtectionDomain().getCodeSource().getLocation().getFile();
        System.err.println("TestNG Version " + new File(file).getParentFile().getName());
        System.err.println("XmlTest name :" + xmlTest.getName());
        System.err.println("Classes : " + xmlTest.getClasses());
    }

    @Test
    public void testMethod() {
        System.err.println("Test method name ");
    }
}
```

And here's the output
```
[TestNG] Running:
  /Users/krmahadevan/Library/Caches/IntelliJIdea2017.1/temp-testng-customsuite.xml
TestNG Version 6.9.9
XmlTest name :testbed
Classes : [[XmlClass class=com.rationaleemotions.github.issue1446.SuiteLevelInjection]]
Test method name 

===============================================
Default Suite
Total tests run: 1, Failures: 0, Skips: 0
===============================================
```